framework/src/main/java/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -936,7 +968,30 @@ public AnnotationMirror createArrayLenRangeAnnotation(Range range) {
         return createArrayLenAnnotation(ValueCheckerUtils.getLengthsForStringValues(values));
     }
 
-    /** Converts an {@code @ArrayLen} annotation to an {@code @ArrayLenRange} annotation. */
+    /**
+     * Converts an {@code StringVal} annotation to an {@code MatchesRegex} annotation that matches
+     * exactly the string values listed in the {@code StringVal}.
+     *
+     * @param stringValAnno a StringVal annotation
+     * @return an equivalent MatchesReges annotation
+     */
+    /* package-private */ AnnotationMirror convertStringValToMatchesRegex(
+            AnnotationMirror stringValAnno) {
+        List<String> values = getStringValues(stringValAnno);
+        List<@Regex String> valuesAsRegexes = new ArrayList<>();
+        for (String value : values) {
+            // Quote the String, and surround it with start-of-string and end-of-string markers.
+            valuesAsRegexes.add("^" + Pattern.quote(value) + "$");
```

Comment:
Because your implementation uses `matches` rather than `Matcher.find`, you do not need these anchors.  The tests all pass without them.